src/main/java/dev/meyi/bn/gui/SettingsGui.java

```diff
@@ -0,0 +1,114 @@
+
+
+    @Override
+    public void initGui() {
+        super.initGui();
+        int buttonId = 0;
+        for (Module m: BazaarNotifier.modules) {
+            buttonList.add(new GuiButton(buttonId, getButtonX(buttonId) , getButtonY(buttonId), m.getReadableName()));
+            buttonId++;
+        }
+        buttonList.add(new GuiButton(buttonId, getButtonX(buttonId),getButtonY(buttonId),
+                BazaarNotifier.activeBazaar ? "Mod active" : "Mod disabled"));
+        buttonId++;
+        buttonList.add(new GuiButton(buttonId, getButtonX(buttonId), getButtonY(buttonId),
+                BazaarNotifier.config.showChatMessages? "Chat messages enabled" : "Chat messages disabled"));
+        buttonId++;
+        apiKey = new GuiTextField(buttonId, fontRendererObj, getButtonX(buttonId), getButtonY(buttonId), 200, 20);
+        apiKey.setMaxStringLength(40);
+        apiKey.setFocused(true);
+        apiKey.setCanLoseFocus(false);
+        apiKey.setText(BazaarNotifier.config.api);
+        previousApiKey = BazaarNotifier.config.api;
+    }
+
+    @Override
+    protected void actionPerformed(GuiButton Button) throws IOException {
+        if (Button.id < BazaarNotifier.modules.size()) {
+            BazaarNotifier.guiToOpen = "module"+ Button.id;
+        }else if (Button.id ==  BazaarNotifier.modules.size()){
+            BazaarNotifier.activeBazaar ^= true;
+            Button.displayString = BazaarNotifier.activeBazaar ? "Mod active" : "Mod disabled";
+        }else if (Button.id ==  BazaarNotifier.modules.size()+1){
+            BazaarNotifier.config.showChatMessages ^= true;
+            Button.displayString = BazaarNotifier.config.showChatMessages? "Chat messages enabled" : "Chat messages disabled";
+        }
+    }
+
+    @Override
+    public void drawScreen(int mouseX, int mouseY, float partialTicks) {
+        this.drawDefaultBackground();
+        super.drawScreen(mouseX, mouseY, partialTicks);
+        apiKey.drawTextBox();
+    }
+
+    @Override
+    public boolean doesGuiPauseGame() {
+        return false;
+    }
+
+    @Override
+    protected void keyTyped(char typedChar, int keyCode) throws IOException {
+        super.keyTyped(typedChar, keyCode);
+        apiKey.textboxKeyTyped(typedChar, keyCode);
+    }
+
+    @Override
+    public void onGuiClosed() {
+        super.onGuiClosed();
+        try {
+            String key = apiKey.getText();
+            if(key.equals("") || key.equals(previousApiKey)){
+                return;
+            }
+            if(Utils.validateApiKey(key)){
+                BazaarNotifier.config.api = key;
+                Minecraft.getMinecraft().thePlayer.addChatMessage(
+                        new ChatComponentText(BazaarNotifier.prefix + "A new Api-Key has been set"));
+            }else{
+                Minecraft.getMinecraft().thePlayer.addChatMessage(new ChatComponentText(
+                        BazaarNotifier.prefix + "Your API-Key was not saved because it was invalid"));
```

Comment:
Keep the text consistent (don't mix up Api and API capitalizations, just use API). Also, add a period to the end of the sentence.

Also, make sure to add some sort of color to it so it doesn't just end up being white. EnumChatFormatting.RED and EnumChatFormatting.GREEN is what we've been using so far (red for error, green for success)